src/gsy_e/models/strategy/ev_charger.py

```diff
@@ -148,3 +136,34 @@ def _sell_energy_to_spot_market(self):
         if self.grid_integration == GridIntegrationType.UNIDIRECTIONAL:
             return  # Do not sell in unidirectional chargers
         super()._sell_energy_to_spot_market()
+
+    def _distribute_energy_to_sessions(
+        self, time_slot: DateTime, traded_energy: float, buyer_name: str, seller_name: str
+    ):
+        """Distribute energy to/from EV sessions based on trade direction."""
+        is_buyer = buyer_name == self.owner.name
+        is_seller = seller_name == self.owner.name
+
+        if is_buyer:
+            # We bought energy - distribute to EV sessions
+            self._state.ev_sessions_manager.distribute_bought_energy(time_slot, traded_energy)
+        elif is_seller:
+            # We sold energy - distribute from EV sessions
+            self._state.ev_sessions_manager.distribute_sold_energy(time_slot, traded_energy)
+
+    def event_offer_traded(self, *, market_id, trade):
+        """Handle offer trades in ONE_SIDED markets."""
```

Comment:
Just saw this by coincidence, please remove "in ONE_SIDED markets" here. Thanks!